src/overlays/actors/ovl_En_Ma_Yto/z_en_ma_yto.c

```diff
@@ -21,126 +83,1425 @@ const ActorInit En_Ma_Yto_InitVars = {
+    this->actionFunc = EnMaYto_AfterMilkRunDialogueHandler;
+}
+
+void EnMaYto_AfterMilkRunDialogueHandler(EnMaYto* this, GlobalContext* globalCtx) {
+    switch (func_80152498(&globalCtx->msgCtx)) {
+        case 0:
+        case 1:
+        case 2:
+        case 3:
+        case 4:
+        case 6:
+            break;
+
+        case 5:
+            EnMaYto_AfterMilkRunChooseNextDialogue(this, globalCtx);
+            break;
+    }
+
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 5, 0x71C, 0xB6);
+}
+
+void EnMaYto_AfterMilkRunChooseNextDialogue(EnMaYto* this, GlobalContext* globalCtx) {
+    if (func_80147624(globalCtx) != 0) {
+        switch (this->textId) {
+            case 0x33C1:
+                EnMaYto_SetFaceExpression(this, 3, 1);
+                // "Thank you. You were cool back there."
+                func_801518B0(globalCtx, 0x33C2, &this->actor);
+                this->textId = 0x33C2;
+                break;
+
+            case 0x33C2:
+                func_801477B4(globalCtx);
+                EnMaYto_SetupPostMilkRunGiveReward(this);
+                EnMaYto_PostMilkRunGiveReward(this, globalCtx);
+                break;
+        }
+    }
+}
+
+void EnMaYto_SetupPostMilkRunGiveReward(EnMaYto* this) {
+    this->actionFunc = EnMaYto_PostMilkRunGiveReward;
+}
+
+void EnMaYto_PostMilkRunGiveReward(EnMaYto* this, GlobalContext* globalCtx) {
+    if (Actor_HasParent(&this->actor, globalCtx)) {
+        EnMaYto_SetupPostMilkRunExplainReward(this);
+    } else if (INV_CONTENT(ITEM_MASK_ROMANIS) == ITEM_MASK_ROMANIS) {
+        func_800B8A1C(&this->actor, globalCtx, GI_RUPEE_GOLD, 500.0f, 100.0f);
+        this->unk_310 = 2;
+    } else {
+        func_800B8A1C(&this->actor, globalCtx, GI_MASK_ROMANIS, 500.0f, 100.0f);
+        this->unk_310 = 1;
+    }
+}
+
+void EnMaYto_SetupPostMilkRunExplainReward(EnMaYto* this) {
+    this->actionFunc = EnMaYto_PostMilkRunExplainReward;
+}
+
+void EnMaYto_PostMilkRunExplainReward(EnMaYto* this, GlobalContext* globalCtx) {
+    if (func_800B84D0(&this->actor, globalCtx)) {
+        if (this->unk_310 == 1) {
+            // Romani's mask explanation
+            EnMaYto_SetFaceExpression(this, 0, 1);
+            func_801518B0(globalCtx, 0x33C3, &this->actor);
+            this->textId = 0x33C3;
+            // Attempted Cremia Cart Ride
+            gSaveContext.weekEventReg[0xE] |= 0x1;
+            this->unk_310 = 3;
+            func_80151BB4(globalCtx, 0x20);
+            func_80151BB4(globalCtx, 0x1F);
+            func_80151BB4(globalCtx, 6);
+            EnMaYto_SetupPostMilkRunWaitDialogueEnd(this);
+        } else {
+            // You already have the mask
+            EnMaYto_SetFaceExpression(this, 0, 1);
+            func_801518B0(globalCtx, 0x33D0, &this->actor);
+            this->textId = 0x33D0;
+            // Attempted Cremia Cart Ride
+            gSaveContext.weekEventReg[0xE] |= 0x1;
```

Comment:
More of an open question, but do we prefer hex or decimal for weekEventReg indices? The only I've been going decimal is the speedruns docs of weekEventReg flags were in decimal so it was easier to read that way, but I'm fine with either

Reply:
I'd say once we have a clear indication of what each are we might be able to make a enum for things like masks.

Reply:
The up-to-date sheet is using hex, so it may be useful to use hex for those indexes.
https://docs.google.com/spreadsheets/d/14OdWg4eg9tIQFa5AxRWRwVlBqWz0f6bw0RAFMQ58_bo/edit#gid=332652535

Reply:
The sheet I was looking at was outdated. Good to know! I'm happy to use hex then